### DISCUSSION TITLE
Icons for pmtiles (MapboxVectorTileCatalogItem)

### DISCUSSION THREAD

**Discussion by aclel** [TerriaJS/terriajs#7267] [Q&A]
I'd like to change the symbol for some of my pmtiles layers. Is anyone currently aware of any example to do this?

Looking at the mapbox JSON styling docs it seems it's possible with layer type `symbol` using an `icon-image`, but I saw that in protomaps-leaflet there is no reference to `icon-image`.

It'd be really nice to be able to use a default icon set like Maki or similar.

**Comment by techmavengeospatial** [comment]
VECTOR TILES USE SPRITES
SPRITE.JSON AND SPRITE.PNG
ALL POINT MARKERS ARE CONTROLLED VIA THE SPRITE
The GL JSON style references to the sprite

https://docs.mapbox.com/mapbox-gl-js/style-spec/sprite/
http://mapbox.github.io/spritezero/
http://blog.gretchenpeterson.com/archives/4887
https://docs.geotools.org/latest/userguide/extension/mbstyle/spec/sprite.html

https://vtse.arcgis.com/documentation/sprite-editor/
https://developers.arcgis.com/rest/services-reference/enterprise/vector-tile-sprite/

a SPRITE is one image with all markers
there are tools to build Sprites
You can use an Online Vector Tile Style Editor to create/edit your GL JSON Style

if you have access to ArcGIS Pro build a VTPK - use 7zip to extract it and inside you will see all the resources
If you use Extract Package Geoprocessing Tool then you get folder of PBF Tiles and you can package those into MBTILES or GPKG or PMTILES or COMTILES
We developed Tile Utilities to make this easier
 Tile Utilities (convert between formats and clip an area of interest)  https://portfolio.techmaven.net/apps/tile-utilities/
 
Offline Map Data Generator https://offlinedatadownloader.techmaven.net/
https://youtu.be/LdFqcroCaR4?si=54DFVCkR3w2VOdydp

If you are looking for a robust solution for serving data to Terriajs or to https://earthexplorer.techmaven.net  check out
Tile Server https://tileserver.techmaven.net/
Tile Server windows has a self service portal 
Map builder with advanced 2D Map and 3D/4D time enabled maps  ( supports Public or Private maps and data)
(the 3D Map in Tile Server and Geo Data Server is Terriajs)

Serves data from postgis or gpkg geopackage or shapefiles or filegdb as:

 - OGC API FEATURES With CQL FILTERING 
- dynamic PNG raster tiles  with CQL FILTERING
- dynamic PBF/MVT vector tiles with CQL FILTERING 
- coming soon... WMS and geotiff as tiles 

Serves cached map tiles (XYZ,TMS,WMTS)
As vector tiles, raster tiles, terriain-elevation tiles):
from:
- mbtiles
- gpkg
- folder of tiles

Serves static GIS files (KML, GeoJSON, 3DTILES, glb 3d Models, etc)
Supports building forms/collections and supports Real-Time Server-Sent Events and Hooks

 3D Scene Server
Serve 3DTILES and SLPK as SCeneServer URLs
https://3dsceneserver.techmaven.net/
https://www.youtube.com/watch?v=asK7xRfb7YQ&t=10sl
Tile package edition
https://youtu.be/WPzopiQrPcc?si=7ZibDrahKtKoJxp2

Vector To Raster Tiles https://vector2rastertiles.techmaven.net/
